modules/gltf/gltf_document.cpp

```diff
@@ -6468,6 +6468,162 @@ Error GLTFDocument::parse(Ref<GLTFState> state, String p_path, bool p_read_binar
 	return OK;
 }
 
+Error GLTFDocument::parse_buffers(Ref<GLTFState> state, const PackedByteArray &p_bytes) {
+	ERR_FAIL_COND_V(p_bytes.size() < 4, FAILED);
+	Error err;
+	Ref<StreamPeerBuffer> stream;
+	stream.instance();
+	stream->set_data_array(p_bytes);
+	uint32_t magic = stream->get_32();
+	if (magic != 0x46546C67) {
+		return FAILED;
+	}
+	Error err;
+	stream->get_32(); // version
+	stream->get_32(); // length
+
+	uint32_t chunk_length = stream->get_32();
+	uint32_t chunk_type = stream->get_32();
+
+	ERR_FAIL_COND_V(chunk_type != 0x4E4F534A, ERR_PARSE_ERROR); //JSON
+	Vector<uint8_t> json_data;
+	json_data.resize(chunk_length);
+	
+	int32_t len = 0;
+	err = stream->get_partial_data(json_data.ptrw(), chunk_length, len);
+	ERR_FAIL_COND_V(len != chunk_length, ERR_FILE_CORRUPT);
+
+	String text;
+	text.parse_utf8((const char *)json_data.ptr(), json_data.size());
+
+	String err_txt;
+	int err_line;
+	Variant v;
+	err = JSON::parse(text, v, err_txt, err_line);
+	if (err != OK) {
+		_err_print_error("", "", err_line, err_txt.utf8().get_data(), ERR_HANDLER_SCRIPT);
+		return err;
+	}
+
+	state->json = v;
+
+	chunk_length = stream->get_32();
+	chunk_type = stream->get_32();
+
+	if (stream->get_available_bytes()) {
+		ERR_FAIL_COND_V(chunk_type != 0x004E4942, ERR_PARSE_ERROR); //BIN
+		state->glb_data.resize(chunk_length);
+		len = 0;
+		err = stream->get_partial_data(state->glb_data.ptrw(), chunk_length, len);
+		ERR_FAIL_COND_V(len != chunk_length, ERR_FILE_CORRUPT);
+
+		err = OK;
+		if (err) {
+			return FAILED;
+		}
+	}
+
+	// // get file's name, use for scene name if none
+	// state->filename = p_path.get_file().get_slice(".", 0);
+
+	ERR_FAIL_COND_V(!state->json.has("asset"), Error::FAILED);
+
+	Dictionary asset = state->json["asset"];
+
+	ERR_FAIL_COND_V(!asset.has("version"), Error::FAILED);
+
+	String version = asset["version"];
+
+	state->major_version = version.get_slice(".", 0).to_int();
+	state->minor_version = version.get_slice(".", 1).to_int();
+
+	/* STEP 0 PARSE SCENE */
+	err = _parse_scenes(state);
+	if (err != OK)
+		return Error::FAILED;
+
+	/* STEP 1 PARSE NODES */
+	err = _parse_nodes(state);
+	if (err != OK)
+		return Error::FAILED;
+
+	/* STEP 2 PARSE BUFFERS */
+	err = _parse_buffers(state, "");
```

Comment:
_parse_buffers must be modified to guarantee that no URI resolves are done if p_base_path is ""